processor/src/main/resources/org/mapstruct/ap/internal/model/assignment/ArrayCopyWrapper.ftl

```diff
@@ -12,4 +12,8 @@
     <@lib.handleSourceReferenceNullCheck>
         <#if ext.targetBeanName?has_content>${ext.targetBeanName}.</#if>${ext.targetWriteAccessorName}<@lib.handleWrite>Arrays.copyOf( ${sourceLocalVarName}, ${sourceLocalVarName}.length )</@lib.handleWrite>;
     </@lib.handleSourceReferenceNullCheck>
+    <#if !ext.defaultValueAssignment?? && !sourcePresenceCheckerReference?? && !ext.targetBeanName?has_content>else {<#-- the opposite (defaultValueAssignment) case is handeld inside lib.handleSourceReferenceNullCheck -->
```

Comment:
we now add an else branch.. But I was thinking: why don't we initialise the variable with `null` (or do we have to handle primitive cases as well correctly then)? Are we going to encounter this problem more often?

Reply:
I think I tried that initially, but it was a bit complex. However, now that you mention I had a look and we can do it when we initialize the variables. I iterated so many times on the initial solution :).